### PR TITLE
Delete remote rates

### PR DESCRIPTION
**Pull request by martpier**
### What does this PR do?

https://datadoghq.atlassian.net/jira/software/projects/ATI/boards/105?selectedIssue=ATI-2313

We decided to abandon the previous "remote rates" in favour of a simpler remote configuration of agent's target TPS.

This removes code related to these remote rates.

It also removes code that sends an ingestion mechanism code alongside the sampling rates to tracers. This is because samplers currently don't use this information, and we don't think it would actually be useful anymore (more details [here](https://datadoghq.atlassian.net/l/cp/ewMFyvQD)).

### Additional Notes

I labeled as `changelog/no-changelog` as this removes a feature that was never made available to customers. It was used internally by Datadog though, so maybe it is a mistake.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by ahmed-mez — APPROVED** [review]

File: pkg/trace/sampler/prioritysampler_test.go [hunk 1]
```diff
@@ -6,17 +6,14 @@
 package sampler
 
 import (
+	"go.uber.org/atomic"
```
Comment (ajgajg1134) [inline pkg/trace/sampler/prioritysampler_test.go:9]: I think this import list needs to be sorted

**Review by ajgajg1134 — CHANGES_REQUESTED** [review]
This is so much better with all that extra code removed!! Just one tiny nit on the imports

File: pkg/trace/sampler/prioritysampler_test.go (hunk 1 above)
Comment (martpier) [reply pkg/trace/sampler/prioritysampler_test.go]: Fixed :)

(how did you catch that omg)

Comment (ajgajg1134) [reply pkg/trace/sampler/prioritysampler_test.go]: 🦅 👀

**Review by ajgajg1134 — APPROVED** [review]

**Review by BaptisteFoy — APPROVED** [review]

**Pull request merged**